docs/reference/rest-api/common-parms.asciidoc

```diff
@@ -631,6 +631,15 @@ If `true`, the destination must be an <<indices-aliases,index alias>>. Defaults
 `false`.
 end::require-alias[]
 
+tag::bulk-dynamic-template-hints[]
+`dynamic_template_hints`::
+(Optional, map)
+A map of hints from the full name of fields to the name of <<dynamic-templates, dynamic templates>.
+Defaults to an empty map. If a name hint matches a dynamic template, then that template will be applied
+regardless of other match predicates defined in the template. And if a field is defined in the mapping,
+then its name hint won't be used.
+end::bulk-dynamic-template-hints[]
```

Comment:
shall we describe what happens when you associate a field with a non existing template? Based on this, I wonder (sorry!) if we should rename this parameter. `hint` may mislead users into thinking that it is best effort and only applied when possible, which I believe does not correspond with the behaviour. maybe now that we use the template name we could remove the hints suffix and go with 'dynamic_template`? Thoughts on this?

Reply:
Good suggestion. I removed "hint" from this PR entirely in https://github.com/elastic/elasticsearch/pull/69948/commits/d2dfda1c6ab2caf5c1e8cdb5c38ec78105fe6489.